src/app/utils/field-validation/validators/dropdownValidator.ts

```diff
@@ -0,0 +1,45 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+
+import { ProcessedSingleAnswerResponse } from 'src/app/modules/submission/submission.types'
+import { IDropdownField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+
+import { types as myInfoTypes } from '../../../../shared/resources/myinfo'
+
+import { notEmptySingleAnswerResponse } from './common'
+import { isOneOfOptions } from './options'
+
+type DropdownValidator = ResponseValidator<ProcessedSingleAnswerResponse>
+type DropdownValidatorConstructor = (
+  dropdownField: IDropdownField,
+) => DropdownValidator
+
+const makeDropdownValidator: DropdownValidatorConstructor = (dropdownField) => (
+  response,
+) => {
+  // Inject fieldOptions for MyInfo. This is necessary because the
+  // client strips out MyInfo data to keep each form submission lightweight
+  const { myInfo } = dropdownField
+  if (myInfo && myInfo.attr) {
+    const [myInfoField] = myInfoTypes.filter(
+      (type) => type.name === myInfo.attr,
+    )
+    const { fieldOptions } = myInfoField
+    if (fieldOptions) dropdownField.fieldOptions = fieldOptions
+  }
+
+  const { fieldOptions } = dropdownField
```

Comment:
suggest extracting the finding of the correct MyInfo fieldOptions into a separate utility function for reuse, then you can express the logic more succinctly with a ternary op:

```typescript
const validOptions = myInfo?.attr ? getMyInfoOptions(myInfo.attr) : fieldOptions
return isOneOfOptions(validOptions, answer)
  ? right(response)
  : left(`DropdownValidator:\t answer is not a valid dropdown option`)
```

Reply:
more generally, we should always find the correct level of abstraction to achieve clean code. in this example we want to switch between `fieldOptions` depending on whether it is a MyInfo field or not. we don't want a reader of the code to delve into implementation details to find out.